src/mscorlib/src/System/TimeZoneInfo.cs

```diff
@@ -1086,35 +1088,9 @@ private class OffsetAndRule {
         //
         // private ctor
         //
-        [System.Security.SecurityCritical]  // auto-generated
-        private TimeZoneInfo(Win32Native.TimeZoneInformation zone, Boolean dstDisabled) {
-            
-            if (String.IsNullOrEmpty(zone.StandardName)) {
-                m_id = c_localId;  // the ID must contain at least 1 character - initialize m_id to "Local"
-            }
-            else {
-                m_id = zone.StandardName;
-            }
-            m_baseUtcOffset = new TimeSpan(0, -(zone.Bias), 0);
-
-            if (!dstDisabled) {
-                // only create the adjustment rule if DST is enabled
-                Win32Native.RegistryTimeZoneInformation regZone = new Win32Native.RegistryTimeZoneInformation(zone);
-                AdjustmentRule rule = CreateAdjustmentRuleFromTimeZoneInformation(regZone, DateTime.MinValue.Date, DateTime.MaxValue.Date, zone.Bias);
-                if (rule != null) {
-                    m_adjustmentRules = new AdjustmentRule[1];
-                    m_adjustmentRules[0] = rule;
-                }
-            }
-
-            ValidateTimeZoneInfo(m_id, m_baseUtcOffset, m_adjustmentRules, out m_supportsDaylightSavingTime);
-            m_displayName = zone.StandardName;
-            m_standardDisplayName = zone.StandardName;
-            m_daylightDisplayName = zone.DaylightName;
-        }
-
 #if PLATFORM_UNIX
-        private TimeZoneInfo(Byte[] data, Boolean dstDisabled) {
+        private TimeZoneInfo(byte[] data, string id, Boolean dstDisabled)
```

Comment:
could it be a good idea to move the isolated Linux specific code to its own cs file? just to reduce the #ifdefs. I am ok to keep what you have if this will not really make the code cleaner anyway.